### PR TITLE
chore: 11773  Migrated `PlatformState` to become a child of `MerkleStateRoot`.

### LINKED ISSUE #11773: Move the location of `PlatformState` to a different place in the Merkle Tree
The Merkle tree that we have should be re-arranged as follows:

1. Current `MerkleHederaState` becomes a root node
2. `PlatformState` becomes a leaf of this new root. 
3. `State` should be deprecated and moved to `testFixtures` submodule in 0.54 release

To solve optimize the tree for state proofs, n-ary node structure have to be replace with "virtual binary" tree. However, it's outside of the scope of this task. 

Here is [a link](https://lucid.app/lucidchart/066f8739-4751-4062-92f3-fddf4f135308/edit?viewport_loc=-2347%2C-129%2C4608%2C2266%2C9GUEelVY~Cfx&invitationId=inv_58b5a9ed-201d-41e6-9f6d-51b17511ee2a) to full Lucid chart

Depends on #11771

### PR DESCRIPTION
**Pull request by imalygin**
**Description**:

This PR changes the structure of the Merkle tree:

image

`PlatformState` migrates to become a `MerkleStateRoot` child. `MerkleStateRoot` becomes a root of the Merkle tree. 

PR highilghts:
- deprecated `com.swirlds.platform.state.State` class for usage in the production code
- used `MerkleRoot` interface where it's possible to reduce usage of the `com.swirlds.platform.state.State` class
- replaced usage of `com.swirlds.platform.state.State` with `com.swirlds.platform.state.MerkleStateRoot` class where possible

**Related issue(s)**:

Fixes #11773 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by lpetrovic05** [comment]
Also I noticed that `MerkleHederaState` has some workaround for the constructable registry. I think it would be sufficient to add the `@ConstructableIgnored` annotation and get rid of this workaround.

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package com.swirlds.platform.state;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.formatting.TextTable;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.utility.MerkleTreeVisualizer;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import java.util.List;
+
+public class MerkleStateUtils {
```
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java:26]: Can we add a Javadoc to the new class?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Generate a string that describes this state.
+     *
+     * @param hashDepth the depth of the tree to visit and print
+     * @param platformState current platform state
+     * @param state current root node state
+     *
+     */
+    public static String createInfoString(int hashDepth, PlatformState platformState, Hash rootHash, MerkleNode state) {
```
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java:35]: if the code in this module is not exempt from using the `final` keyword or `@NonNull`, maybe here is a good place to start

**Review by mxtartaglia-sl — APPROVED** [review]
Approving, the comment is just a suggestion to take or leave

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java (hunk 2 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java]: I added `final` modifiers and `@NonNull` annotation as you recommended.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java (hunk 1 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java:30]: Sure. I added javadoc.

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by litt3 — APPROVED** [review]

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+            assertTrue(
+                    deletionLatch.await(DEFAULT_UNIT_TEST_SECS, TimeUnit.SECONDS),
+                    "Unit test took longer than the default of 10 seconds. Fix the test or override the wait time.");
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            fail();
+        }
+    }
+
+    public void setAddressBook(AddressBook addressBook) {
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java:264]: Javadoc on public method

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+    protected CountDownLatch deletionLatch = new CountDownLatch(1);
+
+    protected AtomicBoolean released = new AtomicBoolean(false);
+
+    public BlockingSwirldState() {
+        super();
+    }
+
+    public BlockingSwirldState(final PlatformState platformState) {
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java:80]: javadoc public constructor

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+    private CountDownLatch serializationLatch;
+
+    protected volatile boolean allowDeletion = true;
+
+    protected CountDownLatch deletionLatch = new CountDownLatch(1);
+
+    protected AtomicBoolean released = new AtomicBoolean(false);
+
+    public BlockingSwirldState() {
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java:76]: javadoc public constructor

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by litt3 — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java (hunk 3 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java]: Fixed

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java (hunk 4 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java]: Fixed

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java (hunk 5 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/state/BlockingSwirldState.java]: Added javadoc

**Review by litt3 — DISMISSED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @imalygin !

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by litt3 — APPROVED** [review]

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]

**Review by rbarkerSL — APPROVED** [review]

**Pull request merged**